source/common/grpc/async_client_impl.cc

```diff
@@ -14,6 +14,39 @@
 namespace Envoy {
 namespace Grpc {
 
+void Base64EscapeBinHeaders(Http::RequestHeaderMap& headers) {
+  absl::flat_hash_map<absl::string_view, std::string> bin_metadata;
+  headers.iterate([&bin_metadata](const Http::HeaderEntry& header) {
+    if (absl::EndsWith(header.key().getStringView(), "-bin")) {
+      bin_metadata.emplace(header.key().getStringView(),
+                           absl::Base64Escape(header.value().getStringView()));
+    }
```

Comment:
Can we directly remove the original header/value pair, and adds the new header/base-64-value pair here? That way, we can avoid the below for loop.

Reply:
I am wondering can you use setCopy() to replace the value inline, like something as below:

headers.iterate([&bin_metadata](const Http::HeaderEntry& header) {
    if (absl::EndsWith(header.key().getStringView(), "-bin")) {
            headers.setCopy(header.key().getStringView(), absl::Base64Escape(header.value().getStringView()));
   }
}